CodeConverter/CSharp/DeclarationNodeVisitor.cs

```diff
@@ -1607,5 +1612,34 @@ public override async Task<CSharpSyntaxNode> VisitTypeConstraint(VBSyntax.TypeCo
             return SyntaxFactory.TypeConstraint(await node.Type.AcceptAsync<TypeSyntax>(_triviaConvertingExpressionVisitor));
         }
 
+        public override async Task<CSharpSyntaxNode> VisitXmlNamespaceImportsClause(VBSyntax.XmlNamespaceImportsClauseSyntax node)
+        {
+            var identifierName = await node.XmlNamespace.Name.AcceptAsync<IdentifierNameSyntax>(TriviaConvertingDeclarationVisitor);
+            var valueLiteral = await node.XmlNamespace.Value.AcceptAsync<ExpressionSyntax>(TriviaConvertingDeclarationVisitor);
+            var declarator = SyntaxFactory.VariableDeclarator(identifierName.Identifier, null, SyntaxFactory.EqualsValueClause(valueLiteral));
+            return SyntaxFactory.FieldDeclaration(
+                SyntaxFactory.List<AttributeListSyntax>(),
+                SyntaxFactory.TokenList(SyntaxFactory.Token(CSSyntaxKind.InternalKeyword), 
+                                        SyntaxFactory.Token(CSSyntaxKind.StaticKeyword), 
+                                        SyntaxFactory.Token(CSSyntaxKind.ReadOnlyKeyword)),
+                SyntaxFactory.VariableDeclaration(SyntaxFactory.IdentifierName("XNamespace"), SyntaxFactory.SingletonSeparatedList(declarator)));
+        }
+
+        public override async Task<CSharpSyntaxNode> VisitXmlName(VBSyntax.XmlNameSyntax node)
+        {
+            if (node.Prefix == null && node.LocalName.ValueText == "xmlns") {
+                // default namespace             
+                return _xmlImportContext.DefaultIdentifierName;
+            } else if (node.Prefix.Name.ValueText == "xmlns") { 
+                // namespace alias
+                return SyntaxFactory.IdentifierName(node.LocalName.ValueText);
+            } else {
+                // Having this case in VB would cause error BC31187: Namespace declaration must start with 'xmlns'
+                throw new NotImplementedException($"Cannot convert non-xmlns attribute in XML namespace import").WithNodeInformation(node);
+            }
+        }
+
+        public override async Task<CSharpSyntaxNode> VisitXmlString(VBSyntax.XmlStringSyntax node) =>
+            CommonConversions.Literal(node.TextTokens.Aggregate("", (a, b) => a + LiteralConversions.EscapeVerbatimQuotes(b.Text)));
```

Comment:
Not sure of the likely inputs here myself, but if there could be quite a few strings and especially if they could be big, consider StringBuilder

Reply:
Agreed. I guess String.Join could also work, right?

Reply:
Yep, that'd probably be slightly more readable too actually 👍